File: Exercism Codes/Java/Clock.java
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Objects;

class Clock {

    LocalDateTime time;

    Clock(int hours, int minutes) {
        time = LocalDate.now().atStartOfDay().plusHours(hours).plusMinutes(minutes);
    }

    void add(int minutes) {
        time = time.plusMinutes(minutes);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Clock clock = (Clock) o;
        return time.getHour() == clock.time.getHour() && time.getMinute() == clock.time.getMinute();
    }

    @Override
    public int hashCode() {
        return Objects.hash(time.getHour(), time.getMinute());
    }

    @Override
    public String toString() {
        return String.format("%02d:%02d", time.getHour(), time.getMinute());
    }
}

File: Exercism Codes/C#/Clock.cs
using System;
public class Clock
{
    private readonly DateTime time;
    public Clock(int hours, int minutes): this(DateTime.Today.Add(new TimeSpan(hours, minutes, 0))) { }
    public Clock(DateTime time) => this.time = time;
    public Clock Add(int minutesToAdd) => new(time.Add(new TimeSpan(0, minutesToAdd, 0)));
    public Clock Subtract(int minutesToSubtract) => new(time.Subtract(new TimeSpan(0, minutesToSubtract, 0)));
    public override string ToString() => $"{time.Hour:00}:{time.Minute:00}";
    public override bool Equals(object obj) => obj is Clock clock && time.Hour == clock.time.Hour && time.Minute == clock.time.Minute;
    public override int GetHashCode() => HashCode.Combine(time.Hour, time.Minute);
}